### PR TITLE
interfaces/builtin/time-control: allow pps access

### PR DESCRIPTION
**Pull request by kubiko**
Allow `/dev/pps[0-9]*` access for time-control interface

Signed-off-by: Ondrej Kubik <ondrej.kubik@canonical.com>

### CODE REVIEW

File: interfaces/builtin/time_control.go [hunk 1]
```diff
@@ -93,6 +93,11 @@ capability sys_time,
 /sys/class/rtc/*/ rw,
 /sys/class/rtc/*/** rw,
 
+# Allow access to pps
+# https://www.kernel.org/doc/html/latest/driver-api/pps.html
```
Comment (anonymouse64) [inline interfaces/builtin/time_control.go:97]: Hmm it seems like pps can be used for other kinds of device not just time things, not sure if it makes sense here or not

File: interfaces/builtin/time_control.go [hunk 2]
```diff
@@ -123,7 +128,10 @@ bind
 socket AF_NETLINK - NETLINK_AUDIT
 `
 
-var timeControlConnectedPlugUDev = []string{`SUBSYSTEM=="rtc"`}
+var timeControlConnectedPlugUDev = []string{
+  `SUBSYSTEM=="rtc"`,
+  `KERNEL=="pps[0-9]*"`,
```
Comment (anonymouse64) [inline interfaces/builtin/time_control.go:133]: this will require a unit test change I think

File: interfaces/builtin/time_control.go (hunk 1 above)
Comment (anonymouse64) [reply interfaces/builtin/time_control.go:97]: After some consideration I think this is fine to include here as the main usage of PPS is indeed for just setting the system clock time, and that's what it is being used for, so I think it's okay to include this here. Others may have differing opinions however

**Review by anonymouse64 — COMMENTED** [review]
lgtm, but I think tests will need to be adjusted

File: interfaces/builtin/time_control.go (hunk 1 above)
Comment (bboozzoo) [reply interfaces/builtin/time_control.go:97]: All the practical usages of pps I've encountered were actually synchronizing with a reliable time source, usually supplied by GPS. I guess logically, you could feed that to ntpd to that fits into the time-control interface, but that's not necessarily the only consumer. OTOH, we don't have an interface for observing time, so time-control sounds like the only choice?

**Review by alexmurray — APPROVED** [review]
LGTM!

**Comment by kubiko** [comment]
@anonymouse64  I pushed fixes for tests now, that should address fails from before

**Review by mvo5 — APPROVED** [review]
Thank you, this looks fine.

**Pull request merged**